src/rdkafka_partition.c

```diff
@@ -3408,6 +3410,13 @@ rd_kafka_topic_partition_list_update (rd_kafka_topic_partition_list_t *dst,
 
                 d->offset = s->offset;
                 d->err    = s->err;
+                if (s->metadata_size > 0) {
```

Comment:
This will leak memory if the `d->metadata` is set. Above this line, add:
```
if (d->metadata) {
        rd_free(d->metadata);
        d->metadata = NULL;
        d->metadata_size = 0;
}
```

Reply:
Done, thanks.